### DISCUSSION TITLE
Nice changes for feature/ADD-TLS

### DISCUSSION THREAD

**Discussion by xyc0815** [tayler6000/pyVoIP#136] [Ideas]
@tayler6000 I like what you've done in the branch feature/ADD-TLS. But there are four things I would like if've you can also do.

- Remove the big bunch of variables in some call and transfer the data via an object. So it is easier to add functions or values in future. Examples are VoIPPhone and VoIPClient. 
- When you enhance the connectivity like you do now, also please add the possiblity to use proxy connections. If you want, I can help you with that.
- Add IPv6 support.
- Reduce in the SIP message the thinks like subRequest += ... and change the variablen names more python style sub_request.

**Comment by tayler6000** [comment]
Hey @xyc0815, thanks! feature/Add-TLS has certainly gotten out of hand hahaha.  But because I divided up the SIP.py in that branch, it's a pain to merge from development to feature/Add-TLS.  I'm going to try to make a big push to get that branch done so I can merge it with development to work on some of these things.

I would like to ensure proxy support works in 2.0.0, do you know of a good proxy to test with?  Preferably something I can dockerize for standardized testing.  I believe Asterisk is designed in such a way it cannot be used as a proxy, but if I'm wrong please let me know.

I intend to add IPv6, as much as I may despise it, but I don't think it will be in 2.0.0.  But maybe 2.1.0 😉 

I intend to refactor everything to use Python style everything such as sub_request, as mentioned in #12 prior to the release of 2.0.0, but I think that may be during the final touches phase of the development.  I am trying to refactor as I go, however.

**Comment by tayler6000** [comment]
> * Remove the big bunch of variables in some call and transfer the data via an object. So it is easier to add functions or values in future. Examples are VoIPPhone and VoIPClient.

Can you elaborate on what you mean by this, and how you think it compares to #144?
